python/google/protobuf/json_format.py

```diff
@@ -614,6 +614,9 @@ def _ConvertStructMessage(self, value, message):
     if not isinstance(value, dict):
       raise ParseError(
           'Struct must be in a dict which is {0}.'.format(value))
+    # Clear will mark the struct as modified so it will be created even if
+    # there are no values.
+    message.Clear()
```

Comment:
The change looks good to me and thanks for the fix!
Just notice that ListValue also called clear on line 608, but didn't find empty ListValue test. Can you help to add such tests as well.

Reply:
done

Reply:
Hey guys! This change breaks behavior.  Prior to this, structs were deep-merged by parse. Now they are not merged anymore...